sql/mysqld.cc

```diff
@@ -4421,6 +4424,83 @@ a file name for --log-bin-index option", opt_binlog_index_name);
     unireg_abort(MYSQLD_ABORT_EXIT);
   }
 
+  if (opt_libcoredumper)
+  {
+#ifndef HAVE_LIBCOREDUMPER
+    sql_print_warning(
+        "This version of MySQL has not been compiled with "
+        "libcoredumper support, ignoring --coredumper argument");
+#else
+    if (opt_corefile)
+    {
+      sql_print_warning(
+          "Started with --core-file and --coredumper. "
+          "--coredumper will take precedence.");
+    }
+    if (opt_libcoredumper_path != NULL)
+    {
+      /* validate path */
+      if (!is_valid_log_name(opt_libcoredumper_path,
+                             strlen(opt_libcoredumper_path)))
+      {  //filename contain .cnf or .ini on it
+        sql_print_error("Variable --coredumper cannot be set to value %s",
+                        opt_libcoredumper_path);
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      char   libcoredumper_dir[FN_REFLEN];
+      size_t libcoredumper_dir_length;
+      size_t opt_libcoredumper_path_length= strlen(opt_libcoredumper_path);
+      (void)dirname_part(libcoredumper_dir, opt_libcoredumper_path,
+                         &libcoredumper_dir_length);
+
+      if (!libcoredumper_dir_length)
+      {
+        sql_print_error("Error processing --coredumper path: %s",
+                        opt_libcoredumper_path);
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      size_t libcoredumper_file_length=
+          opt_libcoredumper_path_length - libcoredumper_dir_length;
+      if (libcoredumper_file_length == 0)  //path is a directory
+      {
+        libcoredumper_file_length= 19;  //file is set to core.yyyymmddhhmmss
+      }
+      else
+      {
+        libcoredumper_file_length+= 17;  //file gets .yyyymmddhhmmss appended
+      }
+      if (opt_libcoredumper_path_length > FN_REFLEN)
+      {  // path is too long
+        sql_print_error("Variable --coredumper set to a too long path");
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      if (libcoredumper_file_length > FN_LEN)
+      {  // filename is too long
+        sql_print_error("Variable --coredumper set to a too long filename");
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      if (my_access(libcoredumper_dir, F_OK))
+      {
+        sql_print_error(
+            "Directory specified at --coredumper: %s does not exist",
+            libcoredumper_dir);
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      if (my_access(libcoredumper_dir, (F_OK | W_OK)))
+      {
+        sql_print_error(
+            "Directory specified at --coredumper: %s is not writable",
```

Comment:
Same question. The disk could also become full in the meantime.

Reply:
in that case there will an error been reported saying it couldn't write the file:

```
bin/mysqld(my_print_stacktrace+0x4b)[0x56068b4cd6a1]
bin/mysqld(handle_fatal_signal+0x44a)[0x56068aa0d177]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x153c0)[0x7f7dff02c3c0]
/lib/x86_64-linux-gnu/libc.so.6(__poll+0x4f)[0x7f7dfef3896f]
bin/mysqld(_ZN22Mysqld_socket_listener27listen_for_connection_eventEv+0x5b)[0x56068b2a5cbb]
bin/mysqld(_ZN19Connection_acceptorI22Mysqld_socket_listenerE21connection_event_loopEv+0x39)[0x560ca3]
bin/mysqld(_Z11mysqld_mainiPPc+0x156d)[0x56068a9fc4ca]
bin/mysqld(main+0x24)[0x56068a9f220d]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3)[0x7f7dfee4a0b3]
bin/mysqld(_start+0x2e)[0x56068a9f212e]
You may download the Percona Server operations manual by visiting
http://www.percona.com/software/percona-server/. You may find information
in the manual which will help you identify the cause of the crash.
Writing a core file using lib coredumper
PATH: /mnt/ramdisk/core

Error writting coredump: -1 Signal: 11
```